From b924fb26f14b38c5a92d7271b2eba73080cfb337 Mon Sep 17 00:00:00 2001
From: Hrvoje Senjan <hrvoje.senjan@gmail.com>
Date: Tue, 30 Jul 2013 01:02:38 +0200
Subject: [PATCH 1/1] qlocale_icu-no-warning-output.patch

Since openSUSE now has it's own icu so versioning, and
we can't predict what will happen in future, qWarnings
about icu libraries and symbols are now only emmited in
debug builds. This will be usefull regardless of icu versioning.

---
 src/corelib/tools/qlocale_icu.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/corelib/tools/qlocale_icu.cpp b/src/corelib/tools/qlocale_icu.cpp
index c3e2907..ce09677 100644
--- a/src/corelib/tools/qlocale_icu.cpp
+++ b/src/corelib/tools/qlocale_icu.cpp
@@ -84,7 +84,9 @@ bool qt_initIcu(const QString &localeString)
         QLibrary lib(QLatin1String("icui18n"), QLatin1String(U_ICU_VERSION_SHORT));
         lib.setLoadHints(QLibrary::ImprovedSearchHeuristics);
         if (!lib.load()) {
+#ifndef QT_NO_DEBUG
             qWarning() << "Unable to load library icui18n" << lib.errorString();
+#endif
             status = ErrorLoading;
             return false;
         }
@@ -105,7 +107,9 @@ bool qt_initIcu(const QString &localeString)
             ptr_ucol_close = 0;
             ptr_ucol_strcoll = 0;
 
+#ifndef QT_NO_DEBUG
             qWarning("Unable to find symbols in icui18n");
+#endif
             status = ErrorLoading;
             return false;
         }
@@ -114,7 +118,9 @@ bool qt_initIcu(const QString &localeString)
         QLibrary ucLib(QLatin1String("icuuc"), QLatin1String(U_ICU_VERSION_SHORT));
         ucLib.setLoadHints(QLibrary::ImprovedSearchHeuristics);
         if (!ucLib.load()) {
+#ifndef QT_NO_DEBUG
             qWarning() << "Unable to load library icuuc" << ucLib.errorString();
+#endif
             status = ErrorLoading;
             return false;
         }
@@ -131,7 +137,9 @@ bool qt_initIcu(const QString &localeString)
             ptr_u_strToUpper = 0;
             ptr_u_strToLower = 0;
 
+#ifndef QT_NO_DEBUG
             qWarning("Unable to find symbols in icuuc");
+#endif
             status = ErrorLoading;
             return false;
         }
-- 
1.8.3.1

